### ISSUE TITLE
Undefined index: scope_condition in SwitchDeclarationSniff

### ISSUE THREAD

**Issue by PatrickRose** [squizlabs/PHP_CodeSniffer#1558]
I'm on PHPCS 2.* and the `PSR2_Sniffs_ControlStructures_SwitchDeclarationSniff` gives a lot of "Undefined index: scope_condition" on the following code:

```php
<?php

switch ($sContext)
{
        case 'SOMETHING':
        case 'CONSTANT':
                do_something();
                break;
        case 'GLOBAL':
        case 'GLOBAL1':
                do_something();
                // Fall through
        default:
        {
                do_something();
        }
}
```

I also get `DEFAULT statements must be defined using a colon` which looks wrong as well.

**Comment by jrfnl** [comment]
What are the braces doing in the `default` case ? I'm actually surprised they don't give a parse error.
Removing the braces would solve your issue.
```php
        default:
                do_something();
```

**Comment by gsherwood** [comment]
The problem here is that PHPCS originally sees the `:` and the `}` switch closer as the open/close points of the default case.

But then it comes across these curly braces surrounding the content of the default case, so it chooses to make those the open and close points because they are more specific.

You've essentially got two complete open/close points for the default case, and PHPCS has selected the most specific one. This has caused the PSR2 error about not using a colon because PHPCS thinks you've chosen to open the default case with curly braces instead.

The  PHP notices are also a direct result of this, but they are caused by something happening in the tokenizer itself and I'll need to debug that and fix it.

**Comment by gsherwood** [comment]
I've changed the way tokenzing works so that this specific code block is handled properly. This characteristics of this code block that caused the error are that it uses curly braces in the default block, and the case blocks above it share the same closer, which happens to be the closing brace of the switch.

I've only made the change in the 3.x version because I'm not keen on making core tokenizer changes to the stable 2.x branch.

As a workaround, I would suggest removing those curly braces as they are redundant in this case. If your coding standard requires them, I'd suggest moving to PHPCS 3 when I release 3.0.1 with this fix (or grab it from master now).

Thanks for reporting this issue.
